File: POLIMORFISMO/ejercicio3/Matriz.java
package ejercicio3;

public class Matriz {
    private float[][] matriz;
    private static final int TAM = 5;

    // Constructor: matriz identidad
    public Matriz() {
        matriz = new float[TAM][TAM];
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                matriz[i][j] = (i == j) ? 1.0f : 0.0f;
            }
        }
    }

    // Constructor: recibe matriz externa
    public Matriz(float[][] valores) {
        matriz = new float[TAM][TAM];
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                if (valores != null && i < valores.length && j < valores[i].length) {
                    matriz[i][j] = valores[i][j];
                } else {
                    matriz[i][j] = 0.0f;
                }
            }
        }
    }

    // Sumar matrices
    public Matriz sumar(Matriz otra) {
        Matriz resultado = new Matriz();
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                resultado.matriz[i][j] = this.matriz[i][j] + otra.matriz[i][j];
            }
        }
        return resultado;
    }

    // Restar matrices
    public Matriz restar(Matriz otra) {
        Matriz resultado = new Matriz();
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                resultado.matriz[i][j] = this.matriz[i][j] - otra.matriz[i][j];
            }
        }
        return resultado;
    }

    // Comparar igualdad
    public boolean igual(Matriz otra) {
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                if (this.matriz[i][j] != otra.matriz[i][j]) {
                    return false;
                }
            }
        }
        return true;
    }

    // Método para mostrar la matriz
    public void mostrar() {
        for (int i = 0; i < TAM; i++) {
            for (int j = 0; j < TAM; j++) {
                System.out.print(matriz[i][j] + " ");
            }
            System.out.println();
        }
    }
}

File: POLIMORFISMO/ejercicio3/matriz.py
# Clase Matriz 10x10
class Matriz:
	def __init__(self, datos=None):
		self.n = 10
		if datos is None:
			# Matriz identidad
			self.matriz = [[1.0 if i == j else 0.0 for j in range(self.n)] for i in range(self.n)]
		else:
			if len(datos) != self.n or any(len(fila) != self.n for fila in datos):
				raise ValueError("La matriz debe ser de 10x10")
			self.matriz = [[float(datos[i][j]) for j in range(self.n)] for i in range(self.n)]

	def sumar(self, otra):
		resultado = [[self.matriz[i][j] + otra.matriz[i][j] for j in range(self.n)] for i in range(self.n)]
		return Matriz(resultado)

	def restar(self, otra):
		resultado = [[self.matriz[i][j] - otra.matriz[i][j] for j in range(self.n)] for i in range(self.n)]
		return Matriz(resultado)

	def igual(self, otra):
		for i in range(self.n):
			for j in range(self.n):
				if self.matriz[i][j] != otra.matriz[i][j]:
					return False
		return True

	def __str__(self):
		return '\n'.join([' '.join([f"{self.matriz[i][j]:.1f}" for j in range(self.n)]) for i in range(self.n)])

# programa principal

m1 = Matriz()  # Matriz identidad
datos = [[float(i+j) for j in range(10)] for i in range(10)]
m2 = Matriz(datos)
suma = m1.sumar(m2)
resta = m2.restar(m1)
print("Matriz identidad:\n", m1)
print("\nMatriz datos:\n", m2)
print("\nSuma:\n", suma)
print("\nResta:\n", resta)
print("\n¿m1 == m2?", m1.igual(m2))
print("¿m1 == m1?", m1.igual(m1))
